agnpy/absorption/absorption.py

```diff
@@ -443,6 +445,113 @@ def evaluate_tau_blr_mu_s(
         )
         return (prefactor * integral).to_value("")
 
+    @staticmethod
+    def compute_integrand(cubepy_params_array, R_line, mu_s, epsilon_1, epsilon_line):
+        """
+        Computes the integrand for the gamma-gamma absorption model with
+        a spherical shell Broad Line Region (BLR).
+
+        Parameters
+        ----------
+        cubepy_params_array : array-like
+            Array containing integration parameters [mu, phi, log_l].
+        R_line : :class:`~astropy.units.Quantity`
+            Radius of the BLR spherical shell.
+        mu_s : float
+            Cosine of the angle between the blob motion and the jet axis.
+        epsilon_1 : float
+            Dimensionless energy parameter for the incoming photon.
+        epsilon_line : float
+            Dimensionless energy of the emitted line.
+
+        Returns
+        -------
+        float
+            The value of the integrand for the given parameters.
+        """
+        mu = cubepy_params_array[0]
+        phi = cubepy_params_array[1]
+        log_l = cubepy_params_array[2]
+        l = np.exp(log_l)
+        mu_star = mu_star_shell(mu, R_line.to_value('cm'), l)
+        _cos_psi = cos_psi(mu_s, mu_star, phi)
+        x = x_re_shell(mu, R_line.to_value('cm'), l)
+        s = epsilon_1 * epsilon_line * (1 - _cos_psi) / 2
+        integrand = (1 - _cos_psi) * l / x ** 2 * sigma(s).to_value("cm**2")
+        return integrand
+    @staticmethod
+    def evaluate_tau_blr_cubepy(
+        nu,
+        eps_abs,
+        z,
+        mu_s,
+        L_disk,
+        xi_line,
+        epsilon_line,
+        R_line,
+        r
+    ):
+        """Evaluates the gamma-gamma absorption produced by a spherical shell
+        BLR for a general set of model parameters using cubepy
+
+        Parameters
+        ----------
+        nu : :class:`~astropy.units.Quantity`
+            array of frequencies, in Hz, to compute the tau
+            **note** these are observed frequencies (observer frame)
+        z : float
+            redshift of the source
+        mu_s : float
```

Comment:
I think this function is following evaluate_tau_blr that is only valid for mu_s=1 (otherwise integration over path of the photon and over the distance from the black hole is not equivalent), so it should not have it as a parameter  (@cosimoNigro can you also have a look at this?)